zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Paragraph.java

```diff
@@ -310,6 +332,43 @@ protected Object jobRun() throws Throwable {
     }
   }
 
+  private int getMaxParagraphOutput()
+  {
+    return replLoader.getZeppelinConfiguration().getInt(MAX_OUTPUT_KEY, 500000);
+  }
+
+
+  private String createTempFile(String message)
+  {
+    String configuredValue = replLoader.getZeppelinConfiguration()
```

Comment:
Lets go off of java.io.tmpdir instead of TMPDIR env var. You should also put all the files under a temp subdirectory.

Use something like `Files.createTempDirectory("synthesys-notebook")`